Sources/MongoSwift/MongoClient.swift

```diff
@@ -548,8 +548,26 @@ public class MongoClient {
         using connection: Connection? = nil,
         session: ClientSession? = nil
     ) -> EventLoopFuture<T.OperationResult> {
+        guard !self.isClosed else {
```

Comment:
(see comment below this one first)

previously this check happened in the executor, but now that I've added the makeFailedFuture helper I think it makes more sense to have the client do the check.